framework/src/main/java/org/checkerframework/framework/util/AnnotatedTypes.java

```diff
@@ -968,37 +968,31 @@ public static List<AnnotatedTypeMirror> adaptParameters(
             List<? extends ExpressionTree> args) {
         List<AnnotatedTypeMirror> parameters = method.getParameterTypes();
 
-        if (parameters.isEmpty()) {
-            return parameters;
-        }
-
         // Handle anonymous constructors that extend a class with an enclosing type.
         if (method.getElement().getKind() == ElementKind.CONSTRUCTOR
                 && method.getElement().getEnclosingElement().getSimpleName().contentEquals("")) {
             DeclaredType t =
                     TypesUtils.getSuperClassOrInterface(
                             method.getElement().getEnclosingElement().asType(), atypeFactory.types);
-            if (t.getEnclosingType() != null) {
-                if (args.isEmpty()) {
-                    // TODO: ugly hack to attempt to fix mismatch
-                    parameters = parameters.subList(1, parameters.size());
-                } else {
-                    TypeMirror p0tm = parameters.get(0).getUnderlyingType();
-                    // Is the first parameter either equal to the enclosing type?
-                    if (atypeFactory.types.isSameType(t.getEnclosingType(), p0tm)) {
-                        // Is the first argument the same type as the first parameter?
-                        if (!atypeFactory.types.isSameType(TreeUtils.typeOf(args.get(0)), p0tm)) {
-                            // Remove the first parameter.
-                            parameters = parameters.subList(1, parameters.size());
-                        }
-                    }
-                }
-                if (parameters.isEmpty()) {
-                    return parameters;
-                }
+            // Let parameterTypes and arguments match, i.e., size, and we only handle Java versions
```

Comment:
Add some comment to the javadoc of this method what this is doing.

Reply:
Thanks! But I don't quite understand as we already have the revelant javadoc. Do you think the current is not enough?